source/v5/mqtt3_to_mqtt5_adapter.c

```diff
@@ -1597,6 +1481,181 @@ static void s_aws_mqtt_client_connection_5_release(void *impl) {
     aws_ref_count_release(&adapter->external_refs);
 }
 
+static void s_adapter_publish_operation_destroy(struct aws_mqtt3_to_mqtt5_adapter_operation_base *operation) {
+    if (operation == NULL) {
+        return;
+    }
+
+    struct aws_mqtt3_to_mqtt5_adapter_operation_publish *publish_op = operation->impl;
+
+    struct aws_mqtt_client_connection_5_impl *adapter_to_release = NULL;
+    if (publish_op->base.holding_adapter_ref) {
+        adapter_to_release = publish_op->base.adapter;
+    }
+
+    if (publish_op->publish_op != NULL) {
+        publish_op->publish_op->completion_options.completion_callback = NULL;
+        publish_op->publish_op->completion_options.completion_user_data = NULL;
+    }
+
+    aws_mqtt5_operation_release(&publish_op->publish_op->base);
+
+    aws_mem_release(operation->allocator, operation);
```

Comment:
Would we like to release `publish_op` instead of `operation` here as we allocate the `publish_op` in `aws_mqtt3_to_mqtt5_adapter_operation_new_publish`?

Meanwhile, if it was an issue, the tests should catch it. Probably I was wrong here?